core/src/main/java/com/sequenceiq/cloudbreak/service/stackpatch/LoggingAgentAutoRestartPatchService.java

```diff
@@ -0,0 +1,150 @@
+import com.sequenceiq.cloudbreak.orchestrator.host.TelemetryOrchestrator;
+import com.sequenceiq.cloudbreak.orchestrator.model.GatewayConfig;
+import com.sequenceiq.cloudbreak.orchestrator.model.Node;
+import com.sequenceiq.cloudbreak.service.CloudbreakRuntimeException;
+import com.sequenceiq.cloudbreak.service.GatewayConfigService;
+import com.sequenceiq.cloudbreak.service.stack.InstanceMetaDataService;
+import com.sequenceiq.cloudbreak.service.stack.StackImageService;
+import com.sequenceiq.cloudbreak.util.CompressUtil;
+
+@Service
+public class LoggingAgentAutoRestartPatchService extends ExistingStackPatchService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LoggingAgentAutoRestartPatchService.class);
+
+    private static final String AFFECTED_CDP_LOGGING_AGENT_VERSION = "0.2.13";
+
+    private static final String CDP_LOGGING_AGENT_PACKAGE = "cdp-logging-agent";
+
+    @Inject
+    private CompressUtil compressUtil;
+
+    @Inject
+    private TelemetryOrchestrator telemetryOrchestrator;
+
+    @Inject
+    private StackImageService stackImageService;
+
+    @Inject
+    private InstanceMetaDataService instanceMetaDataService;
+
+    @Inject
+    private GatewayConfigService gatewayConfigService;
+
+    @Inject
+    private ClusterComponentConfigProvider clusterComponentConfigProvider;
+
+    @Inject
+    private ClusterBootstrapper clusterBootstrapper;
+
+    @Override
+    public boolean isAffected(Stack stack) {
+        Version affectedVersion = Version.parse(AFFECTED_CDP_LOGGING_AGENT_VERSION);
+        try {
+            boolean affected = false;
+            if (StackType.WORKLOAD.equals(stack.getType())) {
+                Image image = stackImageService.getCurrentImage(stack);
+                Map<String, String> packageVersions = image.getPackageVersions();
+                if (packageVersions.containsKey(CDP_LOGGING_AGENT_PACKAGE)
+                        && Version.parse(packageVersions.get(CDP_LOGGING_AGENT_PACKAGE)).compareTo(affectedVersion) <= 0) {
+                    affected = true;
+                }
+            }
+            return affected;
+        } catch (Exception e) {
+            LOGGER.warn("Image not found for stack " + stack.getResourceCrn(), e);
+            throw new CloudbreakRuntimeException("Image not found for stack " + stack.getResourceCrn(), e);
+        }
+    }
+
+    @Override
+    void doApply(Stack stack) throws ExistingStackPatchApplyException {
+        if (isCmServerReachable(stack)) {
+            try {
+                byte[] currentSaltState = clusterComponentConfigProvider.getSaltStateComponent(stack.getCluster().getId());
+                if (currentSaltState == null) {
+                    String message = "Salt state is empty for stack " + stack.getResourceCrn();
+                    LOGGER.info(message);
+                    throw new ExistingStackPatchApplyException(message);
```

Comment:
Should we retry it in this case? If not, simply return.

Reply:
but that would mean patch was applied right? which would be not true, so retry would make sense, at least we would see that something is not right here. im not sure is this even a valid state, but better than got a nullpointer.